replication/binlogsyncer.go

```diff
@@ -178,6 +178,32 @@ func (b *BinlogSyncer) close() {
 		b.c.SetReadDeadline(time.Now().Add(100 * time.Millisecond))
 	}
 
+	// kill last connection id
+	if b.lastConnectionID > 0 {
+		// Use a new connection to kill the binlog dump
+		addr := ""
+		if strings.Contains(b.cfg.Host, "/") {
+			addr = b.cfg.Host
+		} else {
+			addr = fmt.Sprintf("%s:%d", b.cfg.Host, b.cfg.Port)
+		}
+		tmp, err := client.Connect(addr, b.cfg.User, b.cfg.Password, "", func(c *client.Conn) {
+			c.SetTLSConfig(b.cfg.TLSConfig)
+		})
+		if err == nil {
+			cmd := fmt.Sprintf("KILL %d", b.lastConnectionID)
```

Comment:
the code is duplicated, see https://docs.google.com/document/d/1xC3kennA5Z5kK_fL5LDHM9KhmqIxXyZxqoNrJgfssH4/edit

 can you abstract these to a common function?

Reply:
Fixed